builtin/providers/aws/data_source_aws_ami.go

```diff
@@ -236,10 +236,19 @@ func dataSourceAwsAmiRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	var filteredImages []*ec2.Image
-	if nameRegexOk == true {
+	if nameRegexOk {
 		r := regexp.MustCompile(nameRegex.(string))
 		for _, image := range resp.Images {
-			if r.MatchString(*image.Name) == true {
+			// Check for a very rare case where the response would include no
+			// image name. No name means nothing to attempt a match against,
+			// therefore we are skipping such image.
+			if image.Name == nil || *image.Name == "" {
+				log.Printf("[WARN] Unable to find AMI name to match against "+
+					"for image ID %q owned by %q, nothing to do.",
+					*image.ImageId, *image.OwnerId)
```

Comment:
Given the problem here I'm now feeling paranoid: will `image.OwnerId` ever be null? :grinning:

I'm probably over-thinking it, since it seems pretty unlikely that an image would ever not have an owner, but perhaps we should cut this down to just reporting the image id to reduce the chance that we'll crash here trying to prevent a crash.